### PR TITLE
Null safe streamed state builder

### PR DESCRIPTION
**Pull request by IlyaMax**

### CODE REVIEW

**Comment by IlyaMax** [comment]
This closes #217

File: packages/relation/example/lib/demo_relations.dart [hunk 1]
```diff
@@ -43,11 +43,11 @@ class _DemoRelationsState extends State<DemoRelations> {
 
   Future _load() async {
```
Comment (dkrutskikh) [inline packages/relation/example/lib/demo_relations.dart:44]: @IlyaMax please change type to Future<void>

File: packages/relation/test/builder/streamed_state_builder_test.dart [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    'StreamedStateBuilder nullable test',
+    (tester) async {
+      final testStreamedState = StreamedState<String?>(null);
+      final streamedStateBuilder = StreamedStateBuilder<String?>(
+        streamedState: testStreamedState,
+        builder: (context, data) {
+          return Text(data ?? _defaultString);
+        },
+      );
+      await tester.pumpWidget(makeTestableWidget(streamedStateBuilder));
```
Comment (dkrutskikh) [inline packages/relation/test/builder/streamed_state_builder_test.dart:35]: @IlyaMax you can use pumpWidgetBuilder without custom wrapper

File: packages/relation/test/builder/streamed_state_builder_test.dart [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+import 'package:relation/relation.dart';
+
+import '../test_utils.dart';
+
+const _testNotNullString = 'test';
+const _defaultString = 'def';
+
+void main() {
+  testWidgets(
+    'StreamedStateBuilder nullable test',
```
Comment (dkrutskikh) [inline packages/relation/test/builder/streamed_state_builder_test.dart:26]: @IlyaMax please wrap these tests in group

**Review by dkrutskikh — CHANGES_REQUESTED** [review]

File: packages/relation/test/builder/streamed_state_builder_test.dart (hunk 2 above)
Comment (IlyaMax) [reply packages/relation/test/builder/streamed_state_builder_test.dart:35]: I can't find this method. Please give me more detailed info what should I use.

File: packages/relation/test/builder/streamed_state_builder_test.dart (hunk 3 above)
Comment (IlyaMax) [reply packages/relation/test/builder/streamed_state_builder_test.dart]: +

**Review by dkrutskikh — APPROVED** [review]

**Review by trushchinskii-surf — APPROVED** [review]

**Review by artem-zaitsev — APPROVED** [review]

**Pull request merged**